api/errcode.proto

```diff
@@ -9,80 +9,73 @@ option go_package = "berty.tech/berty/go/pkg/errcode";
+  //----------------
+  // Special errors
+  //----------------
   Undefined = 0; // default value, should never be set manually
 
-  TODO = 666; // indicates that you plan to write a custom error handler later
-  ErrNotImplemented = 777;
-  ErrInternal = 999; // can be used to translate an "unknown" error (without Code), i.e., in gRPC
-
-  //
-  // Generic helpers (try to use a more specific error when possible)
-  //
-
-  ErrInvalidInput = 101;
-  ErrMissingInput = 102;
-  ErrSerialization = 103;
-  ErrDeserialization = 104;
-  ErrSignatureFailed = 105;
-  ErrSignatureVerificationFailed = 106;
-  ErrRandomGenerationFailed = 107;
-  ErrCryptoDecrypt = 108;
-  ErrCryptoEncrypt = 109;
-  ErrCryptoKeyConversion = 110;
-  ErrMissingMapKey = 111;
-  ErrNotAuthorized = 112;
-  ErrUnsupportedKeyType = 113;
-
-  ErrOrbitDBInit = 200;
-  ErrOrbitDBOpen = 201;
-  ErrOrbitDBAppend = 202;
-  ErrOrbitDBSerialization = 203;
-  ErrOrbitDBDeserialization = 204;
-  ErrOrbitDBStoreCast = 205;
-  ErrOrbitDBIndexCast = 206;
-
-
-  //
-  // Berty Protocol (starting at 1001)
-  //
-
-  ErrSigChainNoEntries = 1001;
-  ErrSigChainInvalidEntryType = 1002;
-  ErrSigChainAlreadyInitialized = 1003;
-  ErrSigChainPermission = 1004;
-  ErrSigChainOperationAlreadyDone = 1005;
-  ErrHandshakeNoPayload = 1006;
-  ErrHandshakeInvalidFlow = 1007;
-  ErrHandshakeInvalidFlowStepNotFound = 1008;
-  ErrHandshakeParams = 1009;
-  ErrHandshakeNoAuthReturned = 1010;
-  ErrHandshakeInvalidKeyType = 1011;
-  ErrHandshakeInvalidSignature = 1012;
-  ErrHandshakeSessionInvalid = 1013;
-  ErrHandshakeKeyNotInSigChain = 1014;
-  ErrHandshakeDecrypt = 1015;
-  ErrGroupMemberLogEventOpen = 1020;
-  ErrGroupMemberLogEventSignature = 1021;
-  ErrGroupMemberLogWrongInviter = 1022;
-  ErrGroupMemberUnknownGroupID = 1023;
-  ErrGroupMemberMissingSecrets = 1024;
-  ErrGroupSecretOtherDestMember = 1030;
-  ErrGroupSecretEntryDoesNotExist = 1031;
-  ErrGroupSecretAlreadySentToMember = 1032;
-
-  ErrGroupInvitationCantGenerate = 1033;
-  ErrGroupInvalidType = 1034;
-  ErrMissingGroup = 1035;
-
-  ErrSecretKeyGenerationFailed = 1050;
-
-  ErrPersistencePut = 1060;
-  ErrPersistenceGet = 1061;
-
-  //
-  // Bridge (starting at 2001)
-  //
-
-  ErrBridgeInterrupted = 2001;
-  ErrBridgeNotRunning = 2002;
+  TODO = 666; // indicates that you plan to create an error later
+  ErrNotImplemented = 777; // indicates that a method is not implemented yet
+  ErrInternal = 888; // indicates an unknown error (without Code), i.e. in gRPC
+
+  //----------------
+  // Generic errors
+  //----------------
+
+  // Parameters and I/O errors
+  ErrInvalidInput = 100;
+  ErrMissingInput = 101;
+  ErrSerialization = 102;
+  ErrDeserialization = 103;
+  ErrStreamRead = 104;
+  ErrStreamWrite = 105;
+  ErrMissingMapKey = 106;
+
+  // Crypto errors
+  ErrCryptoRandomGeneration = 200;
+  ErrCryptoKeyGeneration = 201;
+  ErrCryptoNonceGeneration = 202;
+  ErrCryptoSignature = 203;
+  ErrCryptoSignatureVerification = 204;
```

Comment:
```suggestion
  ErrCryptoVerification = 204;
```

???

not sure about this one :)

Reply:
Not sure either ^^

I also hesitated to put only `Verification` instead of `SignatureVerification` but by checking a little on the web, usually we talk about signature verification to contextualize.

@glouvigny can you give us your opinion?

Reply:
Looks like `Verification` in context of cryptography could mean a lot of things besides signatures: https://www.google.com/search?lr=lang_en&q=cryptography+verification

I'll keep it as is if you don't mind. ;)